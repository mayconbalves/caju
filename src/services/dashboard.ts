import axios from 'axios'

export const getAllRegisters = async () => {
  try {
    const response = await axios.get(`${import.meta.env.VITE_API_URL}/registrations`)
    return response.data
  } catch (error) {
    console.error('Error fetching registrations:', error)
    throw error
  }
}

export const deleteRegister = async (id: string) => {
  try {
    const response = await axios.delete(`${import.meta.env.VITE_API_URL}/registrations/${id}`)
    if (response.status === 200) {
      return getAllRegisters()
    }
  } catch (error) {
    console.error('Error deleting registration:', error)
    throw error
  }
}

export const updateRegistrationStatus = async (registration: any, status: string) => {
  try {
    const response = await axios.put(
      `${import.meta.env.VITE_API_URL}/registrations/${registration.id}`,
      {
        ...registration,
        status
      }
    )
    if (response.status === 200) {
      return getAllRegisters()
    }
  } catch (error) {
    console.error('Error updating registration:', error)
    throw error
  }
}

export const getRegistrationByDocumentId = async (documentId: string) => {
  try {
    const response = await axios.get(`${import.meta.env.VITE_API_URL}/registrations`, {
      params: { cpf: documentId }
    })

    console.log('Data from API:', response.data)

    if (response.status === 200) {
      if (response.data.length > 0) {
        return response.data
      } else {
        console.error('No registration found')
        return []
      }
    }
  } catch (error) {
    console.error('Error fetching registration by CPF:', error)
    throw error
  }
}
