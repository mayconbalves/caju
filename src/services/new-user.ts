import axios from 'axios'
import { BASE_URL } from '~/constants'

export const createNewUser = async (params: any) => {
  try {
    const response = await axios.post(`${BASE_URL}/registrations`, params)
    return response
  } catch (error) {
    console.error('Error fetching registrations:', error)
    throw error
  }
}
