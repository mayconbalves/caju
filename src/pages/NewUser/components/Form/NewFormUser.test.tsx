import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { useToast } from '~/contexts/Toast/useToast'
import { createNewUser } from '~/services/new-user'
import NewUserForm from '.'

jest.mock('~/services/new-user')
jest.mock('~/contexts/Toast/useToast')

describe('NewUserForm Component', () => {
  const mockShowToast = jest.fn()

  beforeEach(() => {
    ;(useToast as jest.Mock).mockReturnValue({ showToast: mockShowToast })
    ;(createNewUser as jest.Mock).mockResolvedValue({})
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  test('renders form fields', () => {
    render(<NewUserForm />, { wrapper: MemoryRouter })

    expect(screen.getByLabelText(/Nome/i)).toBeInTheDocument()
    expect(screen.getByLabelText(/Email/i)).toBeInTheDocument()
    expect(screen.getByLabelText(/CPF/i)).toBeInTheDocument()
    expect(screen.getByLabelText(/Data de admissÃ£o/i)).toBeInTheDocument()
    expect(screen.getByRole('button', { name: /Cadastrar/i })).toBeInTheDocument()
  })

  it('should be create snapshot', () => {
    const { container } = render(<NewUserForm />, { wrapper: MemoryRouter })
    expect(container).toMatchSnapshot()
  })
})
