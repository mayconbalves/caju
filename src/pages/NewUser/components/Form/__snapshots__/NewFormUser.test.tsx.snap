// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewUserForm Component should be create snapshot 1`] = `
<div>
  <form>
    <div
      class="sc-gsFSjX nzYms"
    >
      <label>
        Nome
      </label>
      <input
        aria-invalid="false"
        aria-label="Nome"
        class="sc-kAycRU iFzHoK"
        name="employeeName"
        placeholder="Nome"
        value=""
      />
    </div>
    <div
      class="sc-gsFSjX nzYms"
    >
      <label>
        Email
      </label>
      <input
        aria-invalid="false"
        aria-label="Email"
        class="sc-kAycRU iFzHoK"
        name="email"
        placeholder="Email"
        type="email"
        value=""
      />
    </div>
    <div
      class="sc-gsFSjX nzYms"
    >
      <label>
        CPF
      </label>
      <input
        aria-invalid="false"
        aria-label="CPF"
        class="sc-kAycRU iFzHoK"
        name="documentId"
        placeholder="CPF"
        value=""
      />
    </div>
    <div
      class="sc-gsFSjX nzYms"
    >
      <label>
        Data de admissão
      </label>
      <input
        aria-invalid="false"
        aria-label="Data de admissão"
        class="sc-kAycRU iFzHoK"
        name="admissionDate"
        type="date"
        value=""
      />
    </div>
    <button
      bgcolor="#64a98c"
      class="sc-aYaIB dnacNp"
      role="button"
      type="submit"
    >
      Cadastrar
    </button>
  </form>
</div>
`;
